fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1718,6 +1743,35 @@ public static IndexState writeOnlyIfTooManyRecordsForRebuild(long recordCount, b
         });
     }
 
+    private boolean indexStateIsStoredInOldFormat(@Nonnull RecordMetaDataProto.DataStoreInfo storeHeader) {
+        return storeHeader.hasFormatVersion() && storeHeader.getFormatVersion() < SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION;
+    }
+
+    private boolean indexStateNeedsToUpgrade() {
+        return formatVersion >= SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION;
+    }
+
+    private CompletableFuture<Void> upgradeIndexStates() {
+        Transaction tr = ensureContextActive();
+        if (recordStoreStateRef.get() == null) {
+            // The record store state must have been loaded already. Do not attempt to preload it here.
+            throw new UninitializedRecordStoreException("record store state is not loaded");
```

Comment:
This should probably use the `uninitializedStoreException` method to create the exception as it adds additional log keys that are useful.

Reply:
sure.